src/tools/lint-docs/src/lib.rs

```diff
@@ -399,16 +399,11 @@ fn generate_lint_output(
                     let rendered = msg["rendered"].as_str().expect("rendered field should exist");
                     Ok(rendered.to_string())
                 }
-                None => {
-                    let rendered: Vec<&str> =
-                        msgs.iter().filter_map(|msg| msg["rendered"].as_str()).collect();
-                    Err(format!(
-                        "did not find lint `{}` in output of example, got:\n{}",
-                        name,
-                        rendered.join("\n")
-                    )
-                    .into())
-                }
+                None => Err(format!(
+                    "did not find lint `{}` in output of example, got:\n{}",
+                    name, stderr
```

Comment:
This seems fine but is a bit unrelated, perhaps stray change from another PR?

Reply:
Oh, it was intentional. The previous output made it hard to diagnose this error (it was empty). With this change the output is this (which makes it a little more obvious what is wrong, though maybe a little harder to read JSON errors):

```
dyld: Library not loaded: @rpath/librustc_driver-81878901dee97a69.dylib
  Referenced from: /Users/eric/Proj/rust/rust/build/x86_64-apple-darwin/stage1/bin/rustc
  Reason: image not found
```

I can move it to a separate commit?

Reply:
Seems fine either way, but probably best in a separate commit with a bit of explanation (in particular, why we weren't already doing the simple thing).

Reply:
Sure!  Actually, I decided to do it slightly differently.  It now still shows the rendered output, which is much easier to read, and shows the non-JSON output separately.